ledger/complete/wal/checkpoint_v6_reader.go

```diff
@@ -20,8 +20,17 @@ import (
 // ErrEOFNotReached for indicating end of file not reached error
 var ErrEOFNotReached = errors.New("expect to reach EOF, but actually didn't")
 
-// TODO: validate the header file and the sub file that contains the root hashes
-var ReadTriesRootHash = readTriesRootHash
+func ReadTriesRootHash(logger zerolog.Logger, dir string, fileName string) (
+	trieRootsToReturn []ledger.RootHash,
+	errToReturn error,
+) {
+	errToReturn = validateCheckpointFile(logger, dir, fileName)
+	if errToReturn != nil {
+		return nil, errToReturn
+	}
+	return readTriesRootHash(logger, dir, fileName)
```

Comment:
When this private function is used, the validation is not needed, because we will read the checkpoint in full and validate the checkpoint. 

However, when using the public function `ReadTriesRootHash`, we are not reading the checkpoint files, therefore the validation is needed. The validation will validates the checksums of the subfiles to ensure the integrity.